server/service/src/asset/validate.rs

```diff
@@ -55,3 +57,14 @@ pub fn check_reason_matches_status(
         _ => false,
     }
 }
+
+pub fn check_locations_are_assigned(
+    location_ids: Vec<String>,
+    connection: &StorageConnection,
+) -> Result<Vec<AssetInternalLocationRow>, RepositoryError> {
+    Ok(
+        AssetInternalLocationRepository::new(connection).query_by_filter(
+            AssetInternalLocationFilter::new().id(EqualFilter::equal_any(location_ids)),
```

Comment:
Does this have a bug when you're trying to save an asset, and the locations are already assigned to the current asset?

I think you want something like this, to correctly check the location_ids, but also ignore any that are assigned to the asset already.
```suggestion
            AssetInternalLocationFilter::new().location_id(EqualFilter::equal_any(location_ids)).asset_id(EqualFilter::not_equal_all(asset_id)),
```